test/e2e/yurt/iot.go

```diff
@@ -0,0 +1,186 @@
+	}
+
+	createPlatformAdmin := func(version string) {
+		Eventually(func() error {
+			return k8sClient.Delete(ctx, &iotv1alpha2.PlatformAdmin{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      platformAdminName,
+					Namespace: namespaceName,
+				},
+			})
+		}, timeout, 500*time.Millisecond).Should(SatisfyAny(BeNil(), &util.NotFoundMatcher{}))
+
+		testPlatformAdmin := iotv1alpha2.PlatformAdmin{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      platformAdminName,
+				Namespace: namespaceName,
+			},
+			Spec: iotv1alpha2.PlatformAdminSpec{
+				Version:  version,
+				PoolName: nodePoolName,
+			},
+		}
+		Eventually(func() error {
+			return k8sClient.Create(ctx, &testPlatformAdmin)
+		}, timeout, 500*time.Millisecond).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
+	}
+
+	BeforeEach(func() {
+		By("Start to run iot test, clean up previous resources")
+		ctx = context.Background()
+		k8sClient = ycfg.YurtE2eCfg.RuntimeClient
+		timeout = 60 * time.Second
+		platformadminTimeout = 5 * time.Minute
+		testVersions = generateTestVersions()
+		namespaceName = "iot-test-namespace"
+		nodePoolName = util.NodePoolName
+		createNamespace()
+	})
+
+	AfterEach(func() {
+		By("Cleanup resources after test")
+		By(fmt.Sprintf("Delete the entire namespaceName %s", namespaceName))
+		Expect(k8sClient.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(BeNil())
+	})
+
+	for _, testVersion := range testVersions {
+		version := testVersion
+		Describe(fmt.Sprintf("Test the %s version of PlatformAdmin", version), func() {
+			BeforeEach(func() {
+				platformAdminName = "test-platform-admin-" + version
+				createPlatformAdmin(version)
+			})
+
+			AfterEach(func() {
+				By(fmt.Sprintf("Delete the platformAdmin %s", platformAdminName))
+				Expect(k8sClient.Delete(ctx, &iotv1alpha2.PlatformAdmin{ObjectMeta: metav1.ObjectMeta{Name: platformAdminName, Namespace: namespaceName}}, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(BeNil())
+			})
+
+			It(fmt.Sprintf("The %s version of PlatformAdmin should be stable in ready state after it is created", version), func() {
+				By("verify the status of platformadmin")
+				Eventually(func() error {
+					testPlatfromAdmin := &iotv1alpha2.PlatformAdmin{}
+					if err := k8sClient.Get(ctx, types.NamespacedName{Name: platformAdminName, Namespace: namespaceName}, testPlatfromAdmin); err != nil {
+						return err
+					}
+					if testPlatfromAdmin.Status.Ready == true {
+						return nil
+					} else {
+						return fmt.Errorf("The %s version of PlatformAdmin is not ready", version)
+					}
+				}, platformadminTimeout, 5*time.Second).Should(Succeed())
+			})
+
+			It(fmt.Sprintf("The %s version of PlatformAdmin should deploy the corresponding components after it is created", version), func() {
```

Comment:
platformAdminStatus.Ready includes both ComponentAvailableCondition and ConfigmapAvailableCondition. Perhaps we can remove the verification test here.